src/transformers/generation_utils.py

```diff
@@ -383,7 +383,11 @@ def generate(
                 # see if BOS token can be used for decoder_start_token_id
                 if bos_token_id is not None:
                     decoder_start_token_id = bos_token_id
-                elif hasattr(self.config, "decoder") and hasattr(self.config.decoder, "bos_token_id"):
+                elif (
+                    hasattr(self.config, "decoder")
+                    and hasattr(self.config.decoder, "bos_token_id")
+                    and self.config.decoder.bos_token_id is not None
```

Comment:
need one for more check for this

Reply:
(out of scope)
I would be down for a helper method
`determine_decoder_start_token_id` to get this out of the main block.